yarl/_url.py

```diff
@@ -769,25 +769,27 @@ def _normalize_path(cls, path):
         return prefix + "/".join(_normalize_path_segments(segments))
 
     @classmethod
-    def _encode_host(cls, host, human=False):
-        try:
-            ip, sep, zone = host.partition("%")
-            ip = ip_address(ip)
-        except ValueError:
+    def _encode_host(cls, host: str, human: bool = False) -> str:
+        raw_ip, sep, zone = host.partition("%")
+        # IP parsing is slow, so its wrapped in an LRU
+        ip_compressed_version = _ip_compressed_version(raw_ip)
```

Comment:
A better design would be to only store hits since the ip address parse code is quick to reject